Fight UI

QuentinTavernier opened the pull request.

QuentinTavernier committed:
Add files via upload

QuentinTavernier committed:
Create .gitkeep

QuentinTavernier committed:
Add files via upload

QuentinTavernier committed:
Delete .gitkeep

QuentinTavernier committed:
Create .gitkeep

QuentinTavernier committed:
Add files via upload

QuentinTavernier committed:
Delete .gitkeep

QuentinTavernier committed:
Create .gitkeep

QuentinTavernier committed:
Add files via upload

QuentinTavernier committed:
Delete .gitkeep

QuentinTavernier committed:
Systeme de combat

V1

QuentinTavernier committed:
Systeme de combat

V1

QuentinTavernier committed:
test

QuentinTavernier committed:
Create test.js

QuentinTavernier committed:
Delete test.js

QuentinTavernier committed:
test

QuentinTavernier committed:
fin test

QuentinTavernier committed:
delete pages

QuentinTavernier committed:
Update combatSysteme.js

QuentinTavernier committed:
Renommage des variables

QuentinTavernier committed:
Rajout Modals pour Attaque et Sac

QuentinTavernier committed:
Police+Capacities

Changement Police + mise en place des capacities

QuentinTavernier committed:
Animation

Ajout animations Pokemons

QuentinTavernier committed:
Rajout potions

RustyNova016 commented on vue/game/FightPVE.js:
```diff
@@ -0,0 +1,661 @@
+  ScrollView,
+} from "react-native";
+
+const mob = [
```
Pour les formats JSON de pokemon:
https://github.com/RustyNova016/PokeCBA/tree/classe_metier/classes/JSON%20Formats

Pour des pokemons test:
https://github.com/RustyNova016/PokeCBA/tree/dev/tests

QuentinTavernier replied:
Oui au début je n'ai pas mis de descriptions car je faisais juste des tests, après si ça fonctionne sur Expo au final.
Après le français je ne l'utilise que pour le texte que lit l'utilisateur. Sinon toutes mes variables sont en anglais.
Et depuis le départ je ne pull que sur feature/combat sauf erreur de ma part ?

RustyNova016 reviewed (changes requested):
2-3 petit trucs:
- Un titre et une description pour le pull request serait bien pour la prochaine fois. Surtout quand il y a beaucoup de changements
- Je vois que Next.js a été retiré. Trop compliqué? Les pages ne marche pas avec react native? Il faudrat voir si Expo et Webpack n'ont pas de problemes. Sinon il faut regenerer le projet.
- Il y a package.lock, donc NPM a été utilisé. Retire le, supprime node_module, et fait "yarn install" (Yarn est meme plus rapide que NPM)
- On melange français et anglais. Il faudrait mieux etre consistent et choisir une seul langue pour le projet. Je penche sur l'anglais car open source et les tables sont en anglais, mais a voir.
- Il faut mieux pull sur dev en premier avant de pull sur master.

Sinon apart le probleme avec Next, je pense que c'est bon pour le pull.

QuentinTavernier committed:
Modification fonctions et correction bugs

Modification de certaines fonctions et correction de bug concernant l'utilisation des potions

QuentinTavernier committed:
Changement de la couleur de la healthbar

QuentinTavernier committed:
Rajout du choix du Pokemon

RustyNova016 commented on vue/game/FightPVE.js:
```diff
@@ -654,8 +695,8 @@ const styles = new StyleSheet.create({
+    left: 310,
     borderRadius: 5,
   },
 });
```
Je pense que tu peux couper ton code en plusieur fichiers. Au-delà de 500 lignes, c'est signe que tu as beaucoup trop de chose dans le même fichier. En coupant, cela permet une meilleure lisibilité du code, et tu ne te perds pas.

RustyNova016 commented on tools/CalculateDamage.js:
```diff
@@ -0,0 +1,33 @@
+export function CalculateDamage(
```
Cette fonction devra être modifier pour utiliser la classe Mob directement, et être cotée serveur.

QuentinTavernier committed:
Rectification de bug au démarrage

Rajout test ourName dans useEffect et appel useEffect avec ourName

QuentinTavernier committed:
Création des classes en dehors du fichier fightPVE

QuentinTavernier committed:
Rajout des effets de types

Modification des dommages en fonction du type de l'attaque et du type du défenseur

QuentinTavernier committed:
Modification dommage suivant types pokémon

Fonctionne maintenant si le pokémon à deux types

QuentinTavernier committed:
Rajout de textes et choix du Poké pour utilisation Item

Rajout de plus variété de texte + possibilité de choisir sur quel Pokemon on utilise l'ITEM
